packages/google_maps_flutter/google_maps_flutter/lib/src/controller.dart

```diff
@@ -18,6 +18,35 @@ class GoogleMapController {
   /// The mapId for this controller
   final int mapId;
 
+  /// State map of the stream subscriptions for this controller, used internally.
+  ///
+  /// The map stores a [StreamSubscription] as key and a [bool] as value.
+  /// Each [StreamSubscription] represents a subscription to an event from the native side.
+  /// The corresponding [bool] indicates whether the subscription has been canceled `true` or not `false`.
```

Comment:
It's unclear to me why the boolean exists, since once it's been canceled there is no reason to track it any more; just being in a set is already a boolean state.

Is this only for the unit test? If so, the unit test should be testing that the subscription is *actually* canceled, via fake subscription that keep track of `cancel` calls, rather than testing an unrelated property that we just hope corresponds to that. As currently written, it would be easy to break the actual functionality without breaking the tests.